File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> stack;
    
    for(int i=0;i<arr.size();i++){
        if(stack.size()==0){
            stack.push_back(arr[i]);
        }
        else{
            if(stack[stack.size()-1]!=arr[i]){
                stack.push_back(arr[i]);
            }
        }
    }

    return stack;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        Stack<Integer>stck=new Stack<>();
        for(int i=0;i<arr.length;i++){
            if(stck.isEmpty()){
                stck.push(arr[i]);
            }else{
                if(stck.peek()!=arr[i]){
                    stck.push(arr[i]);
                }
            }
        }
        
        int[] answer = new int[stck.size()];
        for(int i=0;i<stck.size();i++){
            answer[i]=stck.get(i);
        }
        return answer;
    }
}